pkg/k8s/informers/informers.go

```diff
@@ -0,0 +1,240 @@
+package informers
+
+import (
+	"github.com/rs/zerolog/log"
+	smiTrafficAccessClient "github.com/servicemeshinterface/smi-sdk-go/pkg/gen/client/access/clientset/versioned"
+	smiTrafficSpecClient "github.com/servicemeshinterface/smi-sdk-go/pkg/gen/client/specs/clientset/versioned"
+	smiTrafficSplitClient "github.com/servicemeshinterface/smi-sdk-go/pkg/gen/client/split/clientset/versioned"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+
+	configClientset "github.com/openservicemesh/osm/pkg/gen/client/config/clientset/versioned"
+	policyClientset "github.com/openservicemesh/osm/pkg/gen/client/policy/clientset/versioned"
+)
+
+var (
+	k8sInformerKeys = []InformerKey{
+		InformerKeyNamespace,
+		InformerKeyService,
+		InformerKeyServiceAccount,
+		InformerKeyPod,
+		InformerKeyEndpoints,
+	}
+
+	smiInformerKeys = []InformerKey{
+		InformerKeyTrafficSplit,
+		InformerKeyTrafficTarget,
+		InformerKeyHTTPRouteGroup,
+		InformerKeyTCPRoute,
+	}
+
+	configInformerKeys = []InformerKey{
+		InformerKeyMeshConfig,
+		InformerKeyMeshRootCertificate,
+	}
+
+	policyInformerKeys = []InformerKey{
+		InformerKeyEgress,
+		InformerKeyIngressBackend,
+		InformerKeyUpstreamTrafficSetting,
+		InformerKeyRetry,
+	}
+)
+
+// InformerCollectionOption is a function that modifies an informer collection
+type InformerCollectionOption func(*InformerCollection)
+
+// NewInformerCollection creates a new InformerCollection
+func NewInformerCollection(meshName string, stop <-chan struct{}, opts ...InformerCollectionOption) (*InformerCollection, error) {
+	ic := &InformerCollection{
+		meshName:  meshName,
+		informers: map[InformerKey]*informer{},
+	}
+
```

Comment:
Maybe add a comment here to clarify what is happening in this loop. Something like
```suggestion

// Select specific informers
```